src/components/App.js

```diff
@@ -16,12 +27,56 @@ class App extends Component {
   //   data.append('description', 'some value user types');
   // };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+    };
+  }
+
   jobStateChanged = job => {
     if (!job || (Object.keys(job).length === 0 && job.constructor === Object)) {
-      return 'Unknown';
+      return <div />;
     }
 
-    return job.status;
+    switch (job.status) {
```

Comment:
You can create a component that wraps MoonLoader and takes job.status as a prop.
Other than that it looks great!

Reply:
Ok. I'll make this change in some refactoring I'm doing now in `fix/stateless-components`. In there I'm refactoring many of the components to be stateless.